File: class_1/multi_threads.java
class MultiThreads implements Runnable {
   private Thread t;
   private int threadNum;
   
   MultiThreads( int num) {
      threadNum = num;
   }
   
   public void run() {
      System.out.printf("Thread-%d run\n", threadNum);
      try{
          // 让线程睡眠一会
          Thread.sleep(100000);
      }catch(Exception e){}
      System.out.printf("Thread-%d exit\n", threadNum);
   }
   
   public void start () {
      if (t == null) {
         t = new Thread (this, "");
         t.start ();
      }
   }
}
 
public class multi_threads {
 
   public static void main(String args[]) {
      int i = 0;
      for(i = 0; i < 1000; i++){
          MultiThreads R1 = new MultiThreads(i);
          R1.start();
      }
   }   
}

File: class_1/multi_threads.c
#include <stdio.h>
#include <pthread.h>
#include <unistd.h>
#include <stdlib.h>

typedef struct{
	int num;
	int delay;
}thread_args;

void *print_time(void *args)
{
	thread_args *a = (thread_args *)args;
	printf("Thread-%d run\n", a->num);
	sleep(a->delay);
	printf("Thread-%d exit\n", a->num);
	return NULL;
}

int main(int argc, char *argv[])
{
	int i = 0;
	for(i = 0; i < 1000; i++){
		thread_args *args = (thread_args *)malloc(sizeof(thread_args));
		args->num = i;
		args->delay = 10000;
		pthread_t pid;
		pthread_create(&pid, NULL, print_time, args);
	}
	sleep(100000);
	return 0;
}
